docs/source/task-launch.rst

```diff
@@ -84,67 +117,122 @@ We can submit tasks from other tasks.  This allows us to make decisions while
+.. _The compute function: https://dask.pydata.org/en/latest/scheduler-overview.html#the-compute-function
+
+``dask.compute``
+~~~~~~~~~~~~~~~~
+
+``dask.compute`` behaves as normal: it submits the functions to the graph,
+optimizes for less bandwidth/computation and gathers the results.
 
 .. code-block:: python
 
-   from distributed import worker_client
+    from distributed import Client
+    from dask import delayed, compute
 
-   def process_all(data):
-       with worker_client() as e:
-           elements = e.scatter(data)
-           futures = e.map(process_element, elements)
-           analysis = e.submit(aggregate, futures)
-           result = analysis.result()
-       return result
+    @delayed
+    def fib(n):
+        if n < 2:
+            return n
+        a, b = compute(fib(n-1), fib(n-2))
+        return a + b
 
-    analysis = e.submit(process_all, data)  # spawns many tasks
+    client = Client()  # to change default dask scheduler
+    assert fib(4).compute() == 3
 
-This approach is somewhat complex but very powerful.  It allows you to spawn
-tasks that themselves act as potentially long-running clients, managing their
-own independent workloads.
+Look at the documentation of ``dask.compute`` for more detail.
 
-Extended Example
-~~~~~~~~~~~~~~~~
+``get_client``
+~~~~~~~~~~~~~~
 
-This example computing the Fibonacci numbers creates tasks that submit tasks
-that submit tasks that submit other tasks, etc..
+``get_client`` is a lower-leve interface than ``worker_client``. It exposes
+controls to ``secede`` and ``rejoin`` which can manage the number of workers on
+a node.
 
 .. code-block:: python
 
-   In [1]: from distributed import Client, worker_client
+    from distributed import Client, get_client, secede, rejoin
 
-   In [2]: client = Client()
+    def fib(n):
+        if n < 2:
+            return n
+        client = get_client()
+        jobs = client.map(fib, [n-1, n-2])
+        secede()  # so we don't take up a scheduling slot
+        return sum(client.gather(jobs))
 
-   In [3]: def fib(n):
-      ...:     if n < 2:
-      ...:         return n
-      ...:     else:
-      ...:         with worker_client() as c
-      ...:             a = c.submit(fib, n - 1)
-      ...:             b = c.submit(fib, n - 2)
-      ...:             a, b = c.gather([a, b])
-      ...:             return a + b
-      ...:
+    client = Client()
+    assert fib(4) == 3
 
-   In [4]: future = e.submit(fib, 100)
+Note that if all nodes submit jobs but none call ``secede`` this will lock the
+cluster and no computation will be performed.
 
-   In [5]: future
-   Out[5]: <Future: status: finished, type: int, key: fib-7890e9f06d5f4e0a8fc7ec5c77590ace>
+``rejoin`` is also available, and may be more advantageous if more work is done
+(and not just a ``return`` statement).
 
-   In [6]: future.result()
-   Out[6]: 354224848179261915075
+``dask.worker_client``
```

Comment:
I would use `worker_client` rather than `dask.worker_client`.  The latter implies that worker_client is available in the dask namespace.